pkg/verify/store/parallel/parallel.go

```diff
@@ -0,0 +1,92 @@
+// Package parallel combines several signature stores in a single store.
+// Signatures are searched in each substore simultaneously until a
+// match is found.
+package parallel
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/cluster-version-operator/pkg/verify/store"
+)
+
+type signatureResponse struct {
+	signature []byte
+	errIn     error
+}
+
+// Store provides access to signatures stored in sub-stores.
+type Store struct {
+	Stores []store.Store
+}
+
+// Signatures fetches signatures for the provided digest.
+func (s *Store) Signatures(ctx context.Context, name string, digest string, fn store.Callback) error {
+	nestedCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	responses := make(chan signatureResponse, len(s.Stores))
+	errorChannelCount := 0
+	errorChannel := make(chan error, 1)
+
+	for i := range s.Stores {
```

Comment:
The parallelism is indirectly limited by current implementations of Store via maxSignatureSearch but do you think it needs an explicit limit here? If not perhaps just a comment.

Reply:
`maxSignatureSearch` limits the number of sig's a single HTTP or file store will return.  Parallelism here is just limited by the number of stores we feed the parallel store.  For OCP builds, that's currently hard-coded to two (GCS and the mirrors).

Reply:
Oh, right, I see that now. I looked right pass the local Store stuct.